contracts/SwapKiwi.sol

```diff
@@ -250,54 +261,58 @@ contract SwapKiwi is Ownable, ERC721Holder, ERC1155Holder {
   * @param swapId ID of the swap that the swap participants want to cancel
   */
   function cancelSwap(uint64 swapId) external {
+    Swap memory swap = _swaps[swapId];
+    delete _swaps[swapId];
+
     require(
-      _swaps[swapId].initiator == msg.sender || _swaps[swapId].secondUser == msg.sender,
+      swap.initiator == msg.sender || swap.secondUser == msg.sender,
       "SwapKiwi: Can't cancel swap, must be swap participant"
     );
-    // return initiator NFTs
-    safeMultipleTransfersFrom(
-      address(this),
-      _swaps[swapId].initiator,
-      _swaps[swapId].initiatorNftAddresses,
-      _swaps[swapId].initiatorNftIds,
-      _swaps[swapId].initiatorNftAmounts
-    );
 
-    if(_swaps[swapId].secondUserNftAddresses.length != 0) {
+    if(swap.initiatorNftAddresses.length > 0) {
+      // return initiator NFTs
+      safeMultipleTransfersFrom(
+        address(this),
+        swap.initiator,
+        swap.initiatorNftAddresses,
+        swap.initiatorNftIds,
+        swap.initiatorNftAmounts
+      );
+    }
+
+    if(swap.secondUserNftAddresses.length > 0) {
       // return second user NFTs
       safeMultipleTransfersFrom(
         address(this),
-        _swaps[swapId].secondUser,
-        _swaps[swapId].secondUserNftAddresses,
-        _swaps[swapId].secondUserNftIds,
-        _swaps[swapId].secondUserNftAmounts
+        swap.secondUser,
+        swap.secondUserNftAddresses,
+        swap.secondUserNftIds,
+        swap.secondUserNftAmounts
       );
     }
 
-    if (_swaps[swapId].initiatorEtherValue != 0) {
-      _etherLocked -= _swaps[swapId].initiatorEtherValue;
-      uint128 amountToTransfer = _swaps[swapId].initiatorEtherValue;
-      _swaps[swapId].initiatorEtherValue = 0;
-      _swaps[swapId].initiator.transfer(amountToTransfer);
+    if (swap.initiatorEtherValue != 0) {
+      _etherLocked -= swap.initiatorEtherValue;
+      uint128 amountToTransfer = swap.initiatorEtherValue;
+      swap.initiatorEtherValue = 0;
```

Comment:
clearing memory and using intermediate variable is not needed anymore since everything is in memory, not storage.